src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.Match.Tests.cs

```diff
@@ -1042,16 +1042,30 @@ public static IEnumerable<object[]> Match_Timeout_Throws_MemberData()
             {
                 if (RegexHelpers.IsNonBacktracking(engine))
                 {
+                    // NonBacktracking provides O(n) matching performance and thus doesn't have all
+                    // the timeout checks the backtracking engines require.
                     continue;
                 }
 
-                // All of the below tests have catastrophic backtracking...
-
                 string a50 = new string('a', 50);
                 string a64 = new string('a', 64);
                 string a100 = new string('a', 100);
+                string a1_000_000 = new string('a', 1_000_000);
                 string b50 = new string('b', 50);
 
+                // These aren't catastrophic backtracking, but they will result in O(M*N) work, evaluating
```

Comment:
should we increase the 1ms below, eg., to 10 or 100ms? Reason being, if it's extremely short, the engine may hardly have gotten into the challenging part of the pattern -- it could for example hit the timeout on its first time check after finishing parsing. realistically, we want to confirm the timeout is respected even if it occurs during the meat of the work.

Reply:
> It could for example hit the timeout on its first time check after finishing parsing

Can you point me to the timeout in the code you're referring to?

Reply:
I'm not looking at the code -- I'm thinking of this as a black box test. Say a future developer added a timeout check immediately after parsing, then we broke the timeout checks within pattern matching of this pattern. This test would likely keep passing.

Reply:
> This test would likely keep passing.

I doubt it... a millisecond is an eternity to get into the core matching routine.  Even if it didn't catch such an error deterministically, it would catch it enough that it would be failing left and right.  And non-determinism is the name of the game with timeouts.

Reply:
Also, keep in mind that Environment.TickCount doesn't have 1ms resolution... it has closer to 15ms resolution.

Reply:
OK